171315156 pausa

WillMDev opened the pull request:
-Ahora Viewer soporta texturas sin Collider. (UI elementos)
-Cada vez que se cambie al estado pausa hay que especificar un owner, setPauseOwner(int ownerID), como changeToState() solo sirve para crear el estado si no lo está, he usado un método exterior para pasar a pausa con un owner (setPauseOwner(int ownerID)).
-Para moverse por el menú se usa el stick derecho (avisadme si quereís cambiarlo al izq) y para aceptar, el botón A.

WillMDev committed:
Primera versión de PauseState

-Ahora Viewer soporta texto
-Añadida Clase GameState en constructor de PlayState también (herencia)
-Añadido PauseState como clase y en el vector de estados

WillMDev committed:
Pausa avanzada (leer descripción)

-Viewer ahora soporta elementos UI que no necesitan Collider, de esta forma evito usar un mundo físico en el estado Pausa.
-Para pasar al estado de Pausa, se utiliza setPauseOwner(int owner) de GameStateMachine.
-Añadidos botones al menú y movimiento modular de selección.

WillMDev committed:
Reorganizado handleInput de PauseState y añadido sprites para sonido

WillMDev committed:
Slider de volumen funcional

WillMDev committed:
Input JustUp preferible ante JustDown en menús, y movidos métodos privados

Puede generar comportamientos no queridos Presionar la A y mantenerla cuando se quita la pausa y el jugador se agarra, por eso mejor usar JustUp al darle a un botón del menú.

eljorgemo reviewed (changes requested):
Faltan el ángulo y el clip en el Viewer para elementos de la UI (Viewer.cpp, línea 59)

WillMDev committed:
Añadidos constructores con angulo y clip a Viewer

EstebenR reviewed (changes requested):
El control dentro de la pausa sea o con el stick izquierdo o el d-pad
Also si estas en pausa que si le das al botón otra vez se des-pause

WillMDev committed:
Cambio a stick izquierdo en Pausa y salir con botón Xbox, arreglado input para eje X de sticks

Sigue sin irme el input del D-PAD, si se arreglan lo suyo es usarlos en el menu de pausa.

AdriAlvarez00 reviewed (approved):
Cuando implementemos salir hay que acordarse de colocarlo aquí, pero todo bien

EstebenR reviewed (approved):
Ta to gucci
Me gusta lo del slider del volumen

eljorgemo commented:
I'm impressed pero sin impresora
